lib/event_store_http_client/events/write.rb

```diff
@@ -14,14 +14,12 @@ class Write
         dependency :settings
         dependency :client
 
-        def initialize(type, data, stream_name, version)
```

Comment:
I see.

Thinking about a client, though... Some service code will be writing to a limited number of streams. I guess I would have presumed that the client would have an instance of a stream writer (or reader) per stream, ie:

```ruby
account_stream = StreamWriter.build "account"
account_stream.write data, type, version
```